feature/gribi/ate_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go

```diff
@@ -0,0 +1,381 @@
+}
+
+var (
+	dutPort1 = attrs.Attributes{
+		Desc:    "dutPort1",
+		IPv4:    "192.0.2.1",
+		IPv4Len: ipv4PrefixLen,
+		IPv6:    "2001:0db8::192:0:2:1",
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	atePort1 = attrs.Attributes{
+		Name:    "atePort1",
+		IPv4:    "192.0.2.2",
+		IPv4Len: ipv4PrefixLen,
+		IPv6:    "2001:0db8::192:0:2:2",
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	dutPort2 = attrs.Attributes{
+		Desc:    "dutPort2",
+		IPv4:    "192.0.2.5",
+		IPv4Len: ipv4PrefixLen,
+		IPv6:    "2001:0db8::192:0:2:5",
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	atePort2 = attrs.Attributes{
+		Name:    "atePort2",
+		IPv4:    "192.0.2.6",
+		IPv4Len: ipv4PrefixLen,
+		IPv6:    "2001:0db8::192:0:2:6",
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	dutPort3 = attrs.Attributes{
+		Desc:    "dutPort3",
+		IPv4:    "192.0.2.9",
+		IPv4Len: ipv4PrefixLen,
+		IPv6:    "2001:0db8::192:0:2:9",
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	atePort3 = attrs.Attributes{
+		Name:    "atePort3",
+		IPv4:    "192.0.2.10",
+		IPv4Len: ipv4PrefixLen,
+		IPv6:    "2001:0db8::192:0:2:a",
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	dutPort4 = attrs.Attributes{
+		Desc:    "dutPort4",
+		IPv4:    "192.0.2.13",
+		IPv4Len: ipv4PrefixLen,
+		IPv6:    "2001:0db8::192:0:2:D",
+		IPv6Len: ipv6PrefixLen,
+	}
+
+	atePort4 = attrs.Attributes{
+		Name:    "atePort4",
+		IPv4:    "192.0.2.14",
+		IPv4Len: ipv4PrefixLen,
+		IPv6:    "2001:0db8::192:0:2:E",
+		IPv6Len: ipv6PrefixLen,
+	}
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// configureATE configures ports on the ATE.
+func configureATE(t *testing.T, ate *ondatra.ATEDevice) *ondatra.ATETopology {
+	top := ate.Topology().New()
+
+	p1 := ate.Port(t, "port1")
+	i1 := top.AddInterface(atePort1.Name).WithPort(p1)
+	i1.IPv4().
+		WithAddress(atePort1.IPv4CIDR()).
+		WithDefaultGateway(dutPort1.IPv4)
+	i1.IPv6().
+		WithAddress(atePort1.IPv6CIDR()).
+		WithDefaultGateway(dutPort1.IPv6)
+
+	p2 := ate.Port(t, "port2")
+	i2 := top.AddInterface(atePort2.Name).WithPort(p2)
+	i2.IPv4().
+		WithAddress(atePort2.IPv4CIDR()).
+		WithDefaultGateway(dutPort2.IPv4)
+	i2.IPv6().
+		WithAddress(atePort2.IPv6CIDR()).
+		WithDefaultGateway(dutPort2.IPv6)
+
+	p3 := ate.Port(t, "port3")
+	i3 := top.AddInterface(atePort3.Name).WithPort(p3)
+	i3.IPv4().
+		WithAddress(atePort3.IPv4CIDR()).
+		WithDefaultGateway(dutPort3.IPv4)
+	i3.IPv6().
+		WithAddress(atePort3.IPv6CIDR()).
+		WithDefaultGateway(dutPort3.IPv6)
+
+	p4 := ate.Port(t, "port4")
+	i4 := top.AddInterface(atePort4.Name).WithPort(p4)
+	i4.IPv4().
+		WithAddress(atePort4.IPv4CIDR()).
+		WithDefaultGateway(dutPort4.IPv4)
+	i4.IPv6().
+		WithAddress(atePort4.IPv6CIDR()).
+		WithDefaultGateway(dutPort4.IPv6)
+
+	return top
+}
+
+// configureDUT configures port1, port2, port3 and port4 on the DUT.
+func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
+	d := dut.Config()
+
+	p1 := dut.Port(t, "port1")
+	i1 := &telemetry.Interface{Name: ygot.String(p1.Name())}
+	d.Interface(p1.Name()).Replace(t, configInterfaceDUT(i1, &dutPort1))
```

Comment:
There is a helper in the attrs library that will simplify this a bit and get rid of configInterfaceDUT:

`d.Interface(p1.Name()).Replace(t, dutPort1.NewInterface(p1.Name()))`

The helper looks like it does the same thing as configInterfaceDUT(), but it also covers some deviations for some devices with bugs.